### ISSUE TITLE
sql: impossible to specify odoc with orecord as an arg of a func

### ISSUE THREAD

**Issue by host6** [voedger/voedger#837]
## Problem

**Now, in current architecture**
```sql
TABLE odoc1 INHERITS ODoc (
	orecord1 TABLE orecord1(
		orecord2 TABLE orecord2()
	)
);
EXTENSION ENGINE BUILTIN (
	COMMAND CmdODocOne(odoc1);
);
```
result - `odoc1 undefined`
```sql
-- Deprecated
TYPE DashboardResult (
	HourlySales TABLE HourlySales (
		Hour text NOT NULL,
		Total int64 NOT NULL,
		Count int32 NOT NULL
	)
);
EXTENSION ENGINE BUILTIN (
	QUERY Dashboard RETURNS DashboardResult;
);
```
result - `nested tables not supported in types`
- Problem (hot): 
  - impossible to define e.g. `COMMAND Order(odoc.air.orders)` in sql
  - impossible to specify a container for a func result
    - solved by #915 
```[tasklist]
### Tasks
- [ ] https://github.com/voedger/voedger/issues/908
```

**Comment by mail2nnv** [comment]
Do I understand correctly, that is necessary to use not only `Object`, but also `ODoc` as parameters and results of functions?

That is, at the functional design level, changes to `appdef`:

```go
type IFunction interface {
	IType

	// Parameter. Returns nil if not assigned
        // If assigned must be IObject or IODoc
	Param() IStructure

	// Result. Returns nil if not assigned
        // If assigned must be IObject or IODoc
	Result() IStructure
	…
}

type IFunctionBuilder interface {
	IFunction
	ITypeBuilder

	// Sets command parameter. Must be name of an Object or ODoc.
	// If NullQName passed then it means that function has no parameter.
	// If QNameAny passed then it means that parameter may be any Object or ODoc.
	SetParam(QName) IFunctionBuilder

	// Sets command result. Must be name of an Object or ODoc.
	// If NullQName passed then it means that function has no result.
	// If QNameAny passed then it means that result may be any Object or ODoc.
	SetResult(QName) IFunctionBuilder
        …
    }

```

**Comment by host6** [comment]
- Arg, UnloggedArg of a func could be `Object` or `ODoc` 
- `SetResult(QNameANY)` -> result of a func could be `Object` or any known `ODoc`, `CDoc`, `WDoc`. E.g. result of `q.sys.Collection` is func, not QName, because its actual result depends on args: https://github.com/voedger/voedger/blob/b93eec49b0640e894852862cd59fedb595b382e5/pkg/sys/collection/collection_func.go#L18

**Comment by mail2nnv** [comment]
<details><summary>Tech design in `appdef`, finally:</summary>
<p>

```go
type IFunction interface {
	IType

	// Parameter. Returns nil if not assigned
	Param() IType

	// Result. Returns nil if not assigned
	Result() IType
	…
}

type IFunctionBuilder interface {
	IFunction
	ITypeBuilder

	// Sets command parameter. Must be known structural or data type.
	// If NullQName passed then it means that function has no parameter.
	// If QNameAny passed then it means that parameter may be any.
	SetParam(QName) IFunctionBuilder

	// Sets command result. Must be known structural or data type.
	// If NullQName passed then it means that function has no result.
	// If QNameAny passed then it means that result may be any.
	SetResult(QName) IFunctionBuilder
        …
    }
```

</p>
</details>

**Comment by michael-say** [comment]
The second problem is solved by: #915
